lib/json.rb

```diff
@@ -2,11 +2,105 @@ module JSON
   class JSONError < StandardError; end
   class ParserError < JSONError; end
 
+  def self.generate(object)
+    Generator.new(object).generate
+  end
+
   def self.parse(string, **options)
     tokens = Lexer.new(string).tokens
     Parser.new(tokens, **options).parse
   end
 
+  class Generator
+    def initialize(object)
+      @object = object
+      @string = +''
+    end
+
+    def generate
+      generate_element(@object)
+      @string
+    end
+
+    private
+
+    def generate_element(value)
+      case value
+      when NilClass
+        @string << 'null'
+      when TrueClass
+        @string << 'true'
+      when FalseClass
+        @string << 'false'
+      when String, Symbol
+        generate_string(value)
+      when Array
+        generate_array(value)
+      when Hash
+        generate_object(value)
+      else
+        @string << value.to_s
+      end
+    end
+
+    def generate_string(value)
+      @string << '"'
+      if value.is_a?(Symbol)
+        @string << value.to_s
```

Comment:
A symbol can have weird characters too... we should probably escape this like we do strings.

```
irb(main):002:0> sym = :"foo\0bar\n"
=> :"foo\x00bar\n"
irb(main):003:0> sym.to_s
=> "foo\x00bar\n"
```

Reply:
Yep good catch, I've added another commit to implement the escaping for symbols